vsintegration/src/FSharp.Editor/Navigation/FindUsagesService.fs

```diff
@@ -59,50 +60,52 @@ type internal FSharpFindUsagesService
+                        // A file may be part of multiple projects. We are interested in the one own declaration symbol is from.
+                        return spans |> List.tryFind (fun span -> span.Document.Project.AssemblyName = assembly)
+                    | None -> return None
                 } |> liftAsync
+                    
+            let definitionItem =
+                match declarationSpan with 
+                | None -> 
+                    DefinitionItem.CreateNonNavigableItem(
+                        tags,
+                        ImmutableArray.Create(TaggedText(TextTags.Text, symbol.Ident.idText)),
+                        ImmutableArray.Create(TaggedText(TextTags.Assembly, symbolUse.Symbol.Assembly.SimpleName)))
+                | Some span ->
+                    DefinitionItem.Create(tags, ImmutableArray.Create(TaggedText(TextTags.Text, symbol.Ident.idText)), span)
             
-            for definitionItem in definitionItems do
-                do! context.OnDefinitionFoundAsync(definitionItem) |> Async.AwaitTask |> liftAsync
+            do! context.OnDefinitionFoundAsync(definitionItem) |> Async.AwaitTask |> liftAsync
             
             let! symbolUses =
-                match symbolUse.GetDeclarationLocation document with
-                | Some SymbolDeclarationLocation.CurrentDocument ->
-                    checkFileResults.GetUsesOfSymbolInFile(symbolUse.Symbol) |> liftAsync
-                | scope ->
+                match declarationSpan with
+                | Some _ when symbolUse.IsPrivateToFile ->
+                    asyncMaybe {
+                        let! symbolUses = checkFileResults.GetUsesOfSymbolInFile(symbolUse.Symbol) |> liftAsync
+                        return symbolUses |> Array.map (fun x -> document.Project, x)
+                    }
+                | _ ->
                     let projectsToCheck =
-                        match scope with
-                        | Some (SymbolDeclarationLocation.Projects (declProjects, false)) ->
-                            [ for declProject in declProjects do
-                                yield declProject
-                                yield! declProject.GetDependentProjects() ]
-                            |> List.distinct
-                        | Some (SymbolDeclarationLocation.Projects (declProjects, true)) -> declProjects
-                        // The symbol is declared in .NET framework, an external assembly or in a C# project within the solution.
-                        // In order to find all its usages we have to check all F# projects.
-                        | _ -> Seq.toList document.Project.Solution.Projects
+                        match declarationSpan with
+                        | Some declSpan when symbolUse.Symbol.IsInternalToProject ->
+                            [declSpan.Document.Project]
```

Comment:
Yes, InternalsVisibleTo is ignored. Rename does not work well even for single project, so there is no point to break it further.